phases/13-tools-and-protocols/18-mcp-auth-production/code/main.py

```
@@ -0,0 +1,521 @@
+    print("=" * 72)
+    print("PHASE 13 LESSON 18 - MCP AUTH IN PRODUCTION ON iii PRIMITIVES")
+    print("=" * 72)
+
+    print("\n--- step 1: install auth surface (registers HTTP + cron triggers) ---")
+    install_auth_surface()
+
+    print("\n--- step 2: client discovers the authorization server (RFC 8414) ---")
+    client = MockMCPClient(name="Cursor")
+    asm = client.discover()
+    print(f"  client got registration_endpoint={asm['registration_endpoint']}")
+    print(f"  client confirmed S256 PKCE supported")
```

Ruff (0.15.11)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)